build/templates/session.py.mako

```diff
@@ -126,6 +126,8 @@ class ${context_name.title()}(object):
 class Session(object):
     '''${config['session_description']}'''
 
+    __isfrozen = False
```

Comment:
I'm guessing this idea came from:
https://stackoverflow.com/questions/43460776/python-prevent-dynamic-addition-of-variables-in-static-classes

So far we use single _ for private. I'd suggest following that convention and calling this _is_frozen. If the convention needs to change we can do that for the whole thing in a single PR.